haskell-best-practices.md

```diff
@@ -473,6 +473,214 @@ increases boilerplate.
 Add `LANGUAGE StrictData` to modules that contain only plain records that will
 undergo de/serialization (JSON, CSV, Database, etc).
 
+## Testing
+
+### Export Lists
+
+```hs
+-- Bad
+module FooSpec where
+
+-- Good
+module FooSpec
+  ( spec
+  ) where
+```
+
+### Imports
+
+Each project should have a `TestImport` used for all specs. Optionally, a
+`TestImport.IO` version can exist for writing non-App specs in `IO`.
+
+```hs
+-- Bad
```

Comment:
Please add comments pointing to exactly what's bad to make it obvious for me, the dumb reader.